packages/firestore/test/integration/api/database.test.ts

```diff
@@ -1140,12 +1155,18 @@ apiDescribe('Database', (persistence: boolean) => {
     async () => {
       await withTestDoc(persistence, async docRef => {
         const firestore = docRef.firestore;
-        await expect(
-          firestore.clearPersistence()
-        ).to.eventually.be.rejectedWith(
+        const expectedError =
           'Persistence can only be cleared before a Firestore instance is ' +
-            'initialized or after it is terminated.'
-        );
+          'initialized or after it is terminated.';
+        if (usesModularApi()) {
+          // The modular API throws an exception rather than rejecting the
+          // Promise, which matches our overall handling of API call violations.
+          expect(() => firestore.clearPersistence()).to.throw(expectedError);
```

Comment:
If this is the behavior we want, why not change the classic version to match?

Reply:
I think this meets the bar for "breaking change" since it throws instead of returning a rejected Promise.